tools/acorn-optimizer.js

```diff
@@ -696,6 +696,13 @@ function emitDCEGraph(ast) {
               emptyOut(node); // ignore this in the second pass; this does not root
               return;
             }
+            if (value.right.type === 'Literal') {
+              // this is
+              //  var x = Module['x'] = 1234;
+              // this form occurs when global addresses are exported from the
+              // module.  It doesn't constitute a usage.
+              emptyOut(node);
```

Comment:
@kripken can you pay close attention to this change.    I believe that without this all exported globals will appear to be "used" by the JS code.   I don't think this mattered before because we were only considering functions.

Reply:
Looks correct to me.

Please add an assert that this is a number though to be slightly safer,
```suggestion
              assert(typeof value.right.value === 'number');
              emptyOut(node);
```